### PR TITLE
Markdown rendering

### LINKED ISSUE #3302: Preview markdown files
### Is your feature request related to a problem? Please describe.
Markdown files are an easy way of structuring some quick notes.
Although their plain textural representation is already well structured, adapted rendering for phones would allow these files to be even more useful.
That way one could take down notes on the computer, sync them with Nextcloud to the phone and have an easy to read representation on the phone.

### Describe the solution you'd like
The android app is currently displaying text files within the app. It would be nice if it could render Markdown files in preview mode, i.e. similar to HTML.

### Describe alternatives you've considered
There are apps on the Android app store that allow editing and preview of Markdown files, however using them for preview is rather cumbersome compared to previewing Markdown files within the Nextcloud Android app.

### PR DESCRIPTION
**Pull request by AndyScherzinger**
initial rendering for markdown in the TextPreview fragment

Resolves #3302

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky ready to review :)

For testing I used https://github.com/noties/Markwon/blob/master/README.md and uploaded it to my server. Opening it should: download, launch PreviewText fragement and render Markdown. Beare that this only works for Android16+ (thus the checks ans Manifest stuff.... can we pleases finally move past 14+...)

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java [hunk 1]
```diff
@@ -505,4 +522,13 @@ public void run() {
...
         });
     }
+
+    private void setText(TextView textView, String text, OCFile file) {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN
+            && MimeTypeUtil.MIMETYPE_TEXT_MARKDOWN.equals(file.getMimeType())) {
+            textView.setText(getRenderedMarkdownText(getContext(), text));
+        } else {
+            textView.setText(text);
+        }
```
Comment (thomas-mc-work) [inline src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java]: Alternative approach with slightly improved design:

```java
    private void setText(TextView textView, String text, OCFile file) {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN
            && MimeTypeUtil.MIMETYPE_TEXT_MARKDOWN.equals(file.getMimeType())) {
            text = getRenderedMarkdownText(getContext());
        }

        textView.setText(text);
    }
```

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java [hunk 2]
```diff
@@ -505,4 +522,13 @@ public void run() {
...
+    }
```
Comment (thomas-mc-work) [inline src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java]: Alternative approach with slightly improved design:

```java
    private void setText(TextView textView, String text, OCFile file) {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN
            && MimeTypeUtil.MIMETYPE_TEXT_MARKDOWN.equals(file.getMimeType())) {
            text = getRenderedMarkdownText(getContext());
        }

        textView.setText(text);
    }
```

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.java]: This will lead to codacy warnings since you change an input parameter :/

**Pull request merged**